src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Renamer.cs

```diff
@@ -77,15 +77,17 @@ internal sealed class Renamer
 
         public Task RenameAsync(Project project)
         {
-            return GetStrategy()?.RenameAsync(project, _oldFilePath, _newFilePath) ?? Task.CompletedTask;
+            return GetStrategyAsync(project).Result?.RenameAsync(project, _oldFilePath, _newFilePath) ?? Task.CompletedTask;
         }
 
-        private IRenameStrategy GetStrategy()
+        private async Task<IRenameStrategy> GetStrategyAsync(Project project)
         {
             IRenameStrategy[] strategies = new IRenameStrategy[] {
                 new SimpleRenameStrategy(_threadingService, _userNotificationServices, _optionsSettings, _roslynServices)
             };
-            return strategies.FirstOrDefault(s => s.CanHandleRename(_oldFilePath, _newFilePath));
+
+            var compilation = await project.GetCompilationAsync().ConfigureAwait(false);
```

Comment:
Compilation can be null for projects not supporting compilation, you should add a null check.